.devcontainer/Dockerfile

```diff
@@ -1,30 +1,95 @@
-FROM rust:1.75.0-bookworm
+FROM ubuntu:24.04
 
 ARG USERNAME=vscode
 # replace this with your own user id (outside the container)
 # you can retreive your user id with `id -u`
 # this makes it so files created in the container are still owned by the host user (and not root)
-ARG USER_UID=1001 
+ARG USER_UID=1000
 ARG USER_GID=$USER_UID
+ENV DEBIAN_FRONTEND=noninteractive
+ENV TZ=Etc/UTC
 
-RUN export DEBIAN_FRONTEND=noninteractive \
-    && groupadd --gid $USER_GID $USERNAME \
-    && useradd --uid $USER_UID --gid $USER_GID -m $USERNAME \
+# Check if user with UID 1000 already exists, if not create it
+RUN if ! id -u $USER_UID > /dev/null 2>&1; then \
+    groupadd --gid $USER_GID $USERNAME \
+    && useradd --uid $USER_UID --gid $USER_GID -m $USERNAME; \
+    else \
+    usermod -l $USERNAME $(getent passwd $USER_UID | cut -d: -f1) \
+    && groupmod -n $USERNAME $(getent group $USER_GID | cut -d: -f1); \
+    fi \
     && apt-get update \
-    && apt-get install -y sudo \
+    && apt-get install -y sudo tzdata \
     && echo $USERNAME ALL=\(root\) NOPASSWD:ALL > /etc/sudoers.d/$USERNAME \
     && chmod 0440 /etc/sudoers.d/$USERNAME
 
 USER vscode
 
 RUN sudo apt-get update \
     && sudo apt-get install -y \
-        libasound2-dev \
-        protobuf-compiler \
-        build-essential \
-        clang \
-        zsh \
-        htop \
-        jq \
-        tree \
-    && sh -c "$(curl -fsSL https://raw.githubusercontent.com/ohmyzsh/ohmyzsh/master/tools/install.sh)"
\ No newline at end of file
+    libasound2-dev \
```

Comment:
Why would a CI server need `libasound2` :eyes: 

This reminds me of a time a colleague was running a Mac Mini in his office as part of our CI pipeline. I had SSH access to it and learned that you could use the `say` command to make the device speak arbitrary text strings out loud with its built in speaker... I had a lot of fun with that.

Reply:
Luckily that mac mini is now in my basement so I can enable colleagues to have access to it again!

Reply:
1. This is a `.devcontainer` file, so is meant to be run on a dev's local machine. 
2. This wasnt added by this PR, just changed the formatting. Actually in the master all the way back to original commit!
3. If i had to guess why its here is to allow for things like `echo -e "\a"` style notification dings, which would be useful for long running tasks like e2e tests and builds